modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,213 @@
+      body: event.payload.text
+    })
+  }
+
+  async sendImage(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.title,
+      mediaUrl: [event.payload.url]
+    })
+  }
+
+  async sendCarousel(event: sdk.IO.Event) {
+    for (const { subtitle, title, picture, buttons } of event.payload.elements) {
+      const body = `${title}\n\n${subtitle ? subtitle : ''}`
+
+      const options: MessageOption[] = []
+      for (const { type, title, payload, url, text } of buttons) {
+        if (type === 'open_url') {
+          options.push({ label: `${title} : ${url}`, value: undefined, type: 'url' })
+        } else if (type === 'postback') {
+          options.push({ label: title, value: payload, type: 'postback' })
+        } else if (type === 'say_something') {
+          options.push({ label: title, value: text, type: 'say_something' })
+        }
+      }
+
+      const args = { mediaUrl: picture ? [picture] : undefined }
+      await this.sendOptions(event, body, args, options)
+    }
+  }
+
+  async sendChoices(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.quick_replies.map(x => ({
+      label: x.title,
+      value: x.payload,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.text, {}, options)
+  }
+
+  async sendDropdown(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.options.map(x => ({
+      label: x.label,
+      value: x.value,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.message, {}, options)
+  }
+
+  async sendOptions(event: sdk.IO.Event, text: string, args: any, options: MessageOption[]) {
+    const body = `${text}\n\n${options.map(({ label }, idx) => `${idx + 1}. ${label}`).join('\n')}`
+
+    await this.kvs.set(this.getKvsKey(event.target, event.threadId), options, undefined, '10m')
+
+    await this.sendMessage(event, { ...args, body })
+  }
+
+  async sendMessage(event: sdk.IO.Event, args: any) {
+    await this.twilio.messages.create({
+      ...args,
+      provideFeedback: false,
+      from: event.threadId,
+      to: event.target
+    })
+  }
+}
+
+export async function setupMiddleware(bp: typeof sdk, clients: Clients) {
+  bp.events.registerMiddleware({
+    description:
+      'Sends out messages that targets platform = Twilio.' +
+      ' This middleware should be placed at the end as it swallows events once sent.',
+    direction: 'outgoing',
+    handler: outgoingHandler,
+    name: MIDDLEWARE_NAME,
+    order: 100
+  })
+
+  async function outgoingHandler(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    if (event.channel !== 'twilio') {
+      return next()
+    }
+
+    const client: TwilioClient = clients[event.botId]
+    if (!client) {
+      return next()
+    }
+
+    return client.handleOutgoingEvent(event, next)
+  }
+}
+
+export async function removeMiddleware(bp: typeof sdk) {
+  bp.events.removeMiddleware(MIDDLEWARE_NAME)
+}
```

Comment:
Can you move directly bp.events.remove in the intex.ts file ? there's no plus value of moving it here